packages/aws-cdk-lib/aws-stepfunctions/lib/states/state.ts

```diff
@@ -318,16 +318,16 @@ export abstract class State extends Construct implements IChainable {
   /**
    * Add a choice branch to this state
    */
-  protected addChoice(condition: Condition, next: State) {
-    this.choices.push({ condition, next });
+  protected addChoice(condition: Condition, next: State, comment?: string) {
+    this.choices.push({ condition, next, comment });
```

Comment:
Ideally this function would be 

```ts
protected addChoice(choice: ChoiceTransition) {
  this.choices.push(choice);
}
```

but I guess we can't change the function signature now

Reply:
I wonder if we can do this to help make this function signature stable if we ever add more props to `ChoiceTransition`:

```ts
interface ChoiceTransition extends ChoiceTransitionOptions {
  condition;
  next;
}

interface ChoiceTransitionOptions {
  comment?;
}

protected addChoice(condition: Condition, next: State, options: ChoiceTransitionOptions) {}
```

This has 2 benefits:

- using the function would look like

```ts
choice.when(sfn.Condition.stringEquals('$.color', 'BLUE'), handleBlueItem, {
  comment: 'blue item comment',
});
```

which imo makes it clear that we're providing a comment.

- if we add an additional prop to `ChoiceTransition` in the future, like `default` or something, we can easily add it to `ChoiceTransitionOptions`